autoload/neomake/makers/ft/sh.vim

```diff
@@ -4,20 +4,23 @@ function! neomake#makers#ft#sh#EnabledMakers() abort
     return ['sh', 'shellcheck']
 endfunction
 
-" IDEA: could be detected once from "shellcheck --help" (since older versions
-" support zsh, and newer might do so again).
-let s:shellcheck_supported = ['sh', 'bash', 'dash', 'ksh']
-
 function! neomake#makers#ft#sh#shellcheck() abort
-    let args = ['-fgcc']
-    let shebang = matchstr(getline(1), '^#!\s*\zs.*$')
-    if !len(shebang)
-        if index(s:shellcheck_supported, &filetype) != -1
-            let args += ['-s', &filetype]
-        endif
+    let ext = expand('%:e')
+    let shebang = getline(1)
+
+    if ext ==# 'dash' || shebang =~# 'dash'
+        let ft = 'dash'
+    elseif exists('g:is_bash') || ext ==# 'bash' || shebang =~# 'bash'
+        let ft = 'bash'
+    elseif exists('g:is_kornshell') || exists('g:is_posix') || ext ==# 'ksh'
+        \ || shebang =~# 'ksh'
+        let ft = 'ksh'
+    else
+        let ft = 'sh'
     endif
```

Comment:
Well, I could change it to `bash` instead. That's what shellcheck does if `-s` is not specified.

So, what about this logic?

```vim
    if ext ==# 'dash' || shebang =~# 'dash'
        let ft = 'dash'
    elseif exists('g:is_sh') || shebang =~# '\<sh'
        let ft = 'sh'
    elseif exists('g:is_kornshell') || exists('g:is_posix') || ext ==# 'ksh'
        \ || shebang =~# 'ksh'
        let ft = 'ksh'
    else
        let ft = 'bash'
    endif
```

Note that I left out `ext ==# 'sh'` on purpose, because lots of people use `.sh` extension but use if for non-bourne shell scripting.